scripts/src/docker/ethereum.rs

```diff
@@ -38,6 +37,80 @@ pub struct GethInstance {
     pub erc20_contract_address: Address,
 }
 
+struct GethClient {
+    client: Web3<Http>,
+    endpoint_url: String,
+}
+
+impl GethClient {
+    fn new(endpoint: &GethHttpEndpoint) -> anyhow::Result<Self> {
+        let url = endpoint.to_string();
+        let transport =
+            Http::new(&url).context("unable to initialize http transport to ethereum node")?;
+        Ok(GethClient {
+            client: Web3::new(transport),
+            endpoint_url: url,
+        })
+    }
+    async fn fund_ethereum(&self, address: Address) -> anyhow::Result<()> {
+        send_transaction(
+            self.client.clone(),
+            Some(address),
+            30_000,
+            U256::from(1000u128 * 10u128.pow(18)),
+            Vec::new(),
+        )
+        .await
+        .with_context(|| {
+            format!(
+                "failed to send transaction for funding account {:x} with ether to {}",
+                address, self.endpoint_url
+            )
+        })?;
+
+        Ok(())
+    }
+
+    async fn fund_erc20(&self, address: Address, contract_address: Address) -> anyhow::Result<()> {
+        send_transaction(
+            self.client.clone(),
+            Some(contract_address),
+            100_000,
+            U256::from(0u64),
+            Vec::new(),
```

Comment:
You are not passing the correct data here for funding ERC20. Check your diff and see how the old code put `transfer_fn` here.